code/game/objects/structures/extinguisher.dm

```diff
@@ -48,6 +51,27 @@
 		opened = !opened
 	update_icon()
 
+/obj/structure/extinguisher_cabinet/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(opened)
+		if(istype(W, /obj/item/weldingtool))
+			var/obj/item/weldingtool/WT = W
+			if(!WT.remove_fuel(0, user))
+				to_chat(user, "<span class='notice'>You need more welding fuel to complete this task.</span>")
+				return
+			to_chat(user, "<span class='notice'>You begin cutting \the [src] apart...</span>")
+			playsound(loc, cutting_sound, 40, 1)
+			if(do_after(user, 40, 1, target = src))
+				if(!opened || !WT.isOn())
+					return
+				playsound(loc, cutting_sound, 50, 1)
+				visible_message("<span class='notice'>[user] slices apart \the [src].</span>",
+								"<span class='notice'>You cut \the [src] apart with \the [WT].</span>",
+								"<span class='italics'>You hear welding.</span>")
+				var/turf/T = get_turf(src)
+				new material_drop(T)
+				qdel(src)
+				return
```

Comment:
This actually may be needed if we are moving this into the prior proc declaration as I stated above to avoid unintentional interaction attempts following completion. Will advise once you clear up the duplicate proc declaration so I can examine the logic flow.